engine/src/main/java/org/hibernate/validator/internal/engine/validationcontext/ValidationContext.java

```diff
@@ -35,43 +30,34 @@
  * @author Gunnar Morling
  * @author Guillaume Smet
  */
-public interface ValidationContext<T> {
+public interface ValidationContext<T> extends ConstraintTreeValidationContext {
```

Comment:
So about the names...

I think `ConstraintTreeValidationContext` should be `ValidationContext` and that this one should have a different name as it's a specialized one. It's related to bean validation but the issue is that we already have a `BeanValidationContext` and I can't think of a better name for the other one.

Maybe we could have:
```
ValidationContext
   BaseBeanValidationContext
       BeanValidationContext
       ...
```